src/components/Transaction/Transaction.tsx

```diff
@@ -55,6 +56,23 @@ const Transaction = ({
     const isTransfer =
         txType === ITxType.TX_TYPE_CAT_TRANSFER ||
         txType === ITxType.TX_TYPE_STANDARD_TRANSFER
+
+    // NOTE : if the tx is cat transfer, then the first memo is puzzlehash, so do not show it
+    const filteredMemo =
+        memos && txType === ITxType.TX_TYPE_CAT_TRANSFER
+            ? memos.slice(1)
+            : memos
+
+    const decodedMemos = filteredMemo.map((memo) =>
```

Comment:
Using the function from bls-signature library to decodeMemo is better

Reply:
now remove the decode function, the memo from backend would be decoded first